File: src/main/java/com/qlik/demo/java/BookingRepository.java
package com.qlik.demo.java;

import com.qlik.demo.java.model.BookingDocument;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface BookingRepository extends CrudRepository<BookingDocument, String> {

    List<BookingDocument> findByUserId(UUID userId);
}

File: src/main/kotlin/com/qlik/demo/kotlin/BookingRepository.kt
package com.qlik.demo.kotlin

import com.qlik.demo.kotlin.model.BookingDocument
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface BookingRepository : CrudRepository<BookingDocument, String> {

    fun findByUserId(userId: UUID): List<BookingDocument>
}
